lib/async-emitter.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const { iter } = require('@metarhia/common');
+
+class AsyncEmitter {
+  constructor() {
+    this.events = new Map();
+    this.wrappers = new Map();
+  }
+
+  // Add listener
+  //   name <string> event name
+  //   fn <Function> listener
+  on(name, fn) {
+    let event = this.events.get(name);
+    if (!event) {
+      event = new Set();
+      this.events.set(name, event);
+    }
+    event.add(fn);
+  }
+
+  // Add listener
+  //   name <string> event name
+  //   fn <Function> listener
+  // Returns: <Promise> | <undefined>
+  once(name, fn) {
+    if (fn === undefined) {
+      return new Promise(resolve => {
+        this.once(name, resolve);
+      });
+    }
+    const wrapper = (...args) => {
+      this.remove(name, fn);
+      return fn(...args);
+    };
+    this.wrappers.set(fn, wrapper);
+    this.on(name, wrapper);
+    return undefined;
+  }
+
+  // Emit event
+  //   name <string> event name
+  //   args <any[]>
+  // Returns: <Promise> | <undefined>
```

Comment:
It doesn't return undefined, it always returns `Promise`
```suggestion
  // Returns: <Promise>
```

Reply:
Not an async function with `return undefined`?